src/Type/FunctionClosureTypeChangingExtension.php

```diff
@@ -0,0 +1,16 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+
+interface FunctionClosureTypeChangingExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool;
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type;
```

Comment:
I feel like this method signature is not sufficient because the parameter where the closure is passed to is also significat, right? So we also need to know `ParameterReflection` here. Also probably add it to the `*Supported` method too.

Reply:
> I feel like this method signature is not sufficient because the parameter where the closure is passed to is also significat, right?

hmm, not sure about that. We want to overwrite that parameter by supplying our own. And for my use case I did not need it. The test case I wrote here is basically the use case I needed.

Reply:
Let's say you have a method call with three parameters. Each parameter is a callable. You want this extension to override each callable parameter with its own type.

Your current design only works for methods with a single callable parameter, not multiple. Right?

Reply:
ooh yeah, now I understood. We want to be able to select which parameter we want to modify, right? By checking the parameter name or something.

Reply:
Yes 😊

Reply:
Done ea4494680a8aaee57c03530e69375660aea7999e